lib/view_model/node_list/node_create_or_edit_view_model.dart

```diff
@@ -109,11 +114,15 @@ abstract class NodeCreateOrEditViewModelBase with Store {
       state = IsExecutingState();
       if (editingNode != null) {
         await _nodeSource.put(editingNode.key, node);
+      } else if (existingNode != null) {
+        final node = existingNode;
```

Comment:
instead of calling `existingNode` getter twice and looping through the list of nodes twice, we can just have a function that returns a boolean, whether this node exists or not, if it does then just `setAsCurrent`
```dart
else if (isExistingNode(node)) {
     setAsCurrent(node);
}
```
and you can remove the getter above `node` since there will be no use for it and we already have it as the final variable node in this function (line 106)